packages/flutter/lib/src/semantics/semantics.dart

```diff
@@ -2776,6 +2782,95 @@ class SemanticsOwner extends ChangeNotifier {
   String toString() => describeIdentity(this);
 }
 
+/// A string that carries a list of [StringAttribute].
+@immutable
+class AttributedString {
+  /// Created Attributed string.
+  ///
+  /// the [TextRange] in the [attributes] must be in side the length of the
+  /// [string].
+  const AttributedString(
+    this.string, {
+    this.attributes,
+  });
+
+  /// The plain string.
+  final String string;
+
+  /// The attributes this string carries.
+  final List<StringAttribute>? attributes;
+
+  @override
+  bool operator ==(Object other) {
+    if (other.runtimeType != runtimeType)
+      return false;
+    return other is AttributedString
+        && other.string == string
+        && other.attributes == attributes;
+  }
+
+  @override
+  int get hashCode => string.hashCode ^ attributes.hashCode;
+}
+
+/// The attribute type that is used in [StringAttribute].
+enum StringAttributeType {
+  /// The string should be spell out character by character.
+  spellOut,
+}
+
+/// The attribute that affects how assistive technologies treat the text.
+class StringAttribute {
+  /// Creates a text attribtue.
+  ///
+  /// The [range] specifies the portion in the text that carries this attribute
+  ///
+  /// Sets [StringAttributeType.spellOut] to true if you want the voiceover or talkback to spell out
+  /// the text character by character.
+  StringAttribute({
+    required TextRange range,
+    required this.type,
+    this.args,
+  }) : assert(range.isValid && !range.isCollapsed),
+       _range = range;
+
+  /// The text range to which this attribute should apply.
+  TextRange get range => _range;
+  TextRange _range;
+
+  /// The attribute type that this range is carrying.
+  StringAttributeType type;
+
+  /// Additional arguments.
+  Object? args;
+
+  static const String _kTextStartKey = 'StringAttribute_textStartKey';
```

Comment:
Wondering whether we need these keys or if this could just be serialized as a list where we now that the first index is the start, the second one is the end, etc...

Reply:
yes, I think that will work. would that be more preferable in this case??